### PR TITLE
3967 pool leak

### PR DESCRIPTION
**Pull request by zxq9**
Fix memory leak noted in #3967 by converting from arrays to lists.

### CODE REVIEW

File: apps/aecore/src/aec_peers_pool.erl [hunk 1]
```diff
@@ -773,8 +782,8 @@ reference_count(St, PeerId) ->
 find_peer(St, PeerId) ->
     #?ST{peers = Peers} = St,
     case maps:find(PeerId, Peers) of
```
Comment (velzevur) [inline apps/aecore/src/aec_peers_pool.erl:784]: Maybe `maps:get/3` would be a better fit here. Just a note, not a blocker

File: apps/aecore/test/aec_peers_pool_tests.erl [hunk 2]
```diff
@@ -1514,6 +1520,7 @@ validate_counters() ->
 
     % Delete all the peers.
     Pool4 = lists:foldl(fun({I, _}, P) ->
+        erlang:display(P, I),
```
Comment (velzevur) [inline apps/aecore/test/aec_peers_pool_tests.erl:1523]: TIL: `Erlang:display/1` - thanks!

**Review by velzevur — APPROVED** [review]
Great work!

File: apps/aecore/src/aec_peers_pool.erl (hunk 1 above)
Comment (zxq9) [reply apps/aecore/src/aec_peers_pool.erl:784]: That's true all over the module. The structure and choice of calls is pretty bizarre. If you check where `find_peer/2` is called and the way it is called, it is a wrapper around `maps:find/2` that reduces it to the *less* succinct `lists:keyfind/3`. I want to kill all of this weirdness in a cleanup commit later, but there are a *lot* of stupid things like this and I don't have time right now to care about it. In three weeks I will come back and chop all of this nonsense up.

File: apps/aecore/test/aec_peers_pool_tests.erl (hunk 2 above)
Comment (zxq9) [reply apps/aecore/test/aec_peers_pool_tests.erl:1523]: Dhoh! I should remove that. It adds nothing to the pass/fail nature of the tests themselves.

**Review by velzevur — APPROVED** [review]

**Review by uwiger — APPROVED** [review]

**Pull request merged**